### PR TITLE
Closes #605 Update cdn-deploy-main.yml to be triggered on both main and 2.x

### LINKED ISSUE #605: Update cdn-deploy-main.yml to be triggered on both main and 2.x
- [x] Rename cdn-deploy-main.yml to cdn-deploy-head.yml

- [x] Change name of workflows from Build & deploy CDN assets to Build & deploy CDN assets from HEAD

- [x] Change buckets from 
```         
 aws s3 sync --cache-control max-age=691200 dist/ s3://${{ secrets.AZ_DIGITAL_CDN_BUCKET }}/lib/arizona-bootstrap/main/
          aws cloudfront create-invalidation --distribution-id ${{ secrets.AZ_DIGITAL_CDN }} --paths /lib/arizona-bootstrap/main/*
```

https://github.com/az-digital/arizona-bootstrap/blob/2.x/.github/workflows/cdn-deploy-main.yml#L25-L26

```
          aws s3 sync --cache-control max-age=691200 dist/ s3://${{ secrets.AZ_DIGITAL_CDN_BUCKET }}/lib/arizona-bootstrap/${GITHUB_REF}/
          aws cloudfront create-invalidation --distribution-id ${{ secrets.AZ_DIGITAL_CDN }} --paths /lib/arizona-bootstrap/${GITHUB_REF}/*
```

- [ ] Replace references to main bucket in docs to `2.x` on the `2.x` branch.

### PR DESCRIPTION
**Pull request by trackleft**
In this PR, I am changing the github action for deploying merges into main `cdn-deploy-main.yml` to `cdn-deploy-head.yml` and ensuring that both the main and 2.x branches are covered. Additionally, updated the release and readme documentation to account for how the buckets work.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,77 @@
...
+- Update `latest-2x` bucket
+- Create a new bucket for the version that was released. Example: `2.0.20`
+
+### Pull request is merged into `main` branch
+All code merged into the `main` branch will update `5.x` bucket.
+
+### Pull request is merged into `2.x` branch
+All code merged into the `2.x` branch will update the `2.x` bucket.
+
+
```
Comment (trackleft) [inline README.md:61]:
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -1,13 +1,76 @@
...
+### Pull request is merged into `2.x` branch
+All code merged into the `2.x` branch will update the `2.x` bucket.
+
+#### The `<asset type>` parameter can be any of the following
+- `css`
+- `js`
+
+#### The `<filename>` parameter is the specific file you are attempting to use.
+
```
Comment (trackleft) [inline README.md:66]:
```suggestion
```

**Comment by trackleft** [comment]
I think we will need to determine whether these changes to our architecture are appropriate, and if so, let developers know that we will no longer be updating the main bucket on the CDN.

I recommend that we deprecate the main bucket because of the starter template within our docs that calls for it. https://digital.arizona.edu/arizona-bootstrap/docs/2.0/getting-started/introduction/#starter-template

**Comment by trackleft** [comment]
I also recommend the same treatment for the `latest` bucket, deprecate and use another name. 
Possibilities to consider:
`latest-2x`
`latest-5.1x`
`latest-5.2x`
`latest-5.x`

**Comment by mmunro-ltrr** [comment]
The S3 bucket name changes sound right.

File: README.md [hunk 3]
```diff
@@ -1,13 +1,75 @@
...
+#### The `<version bucket>` parameter can be any of the following
+- `main` Deprecated
+- `2.x` Development version of Arizona Bootstrap 2
+- `5.x` Development version of Arizona Bootstrap 5
+- A release version number, Example: `2.0.19` See [the release
+page](https://github.com/az-digital/arizona-bootstrap/releases) for a full list
+of possible options. Remove the `v` from the version number.
+- `latest-2x` The latest stable version of Arizona Bootstrap 2
+- `latest-5x` The latest stable version of Arizona Bootstrap 5
```
Comment (trackleft) [inline README.md:36]:
```suggestion
- `latest-2.x` The latest stable version of Arizona Bootstrap 2
- `latest-5.x` The latest stable version of Arizona Bootstrap 5
```

File: README.md [hunk 4]
```diff
@@ -1,13 +1,75 @@
...
+- `latest` Deprecated
+
+**NOTE: `main` is deprecated and will no longer be updated as of `v2.0.20` - to
+use the latest development version of the `2.x` branch use `2.x`**
+
+**NOTE: `latest` is deprecated and will no longer be updated as of `v2.0.20` -
+to use the latest stable version of the `2.x` branch use `latest-2x`**
```
Comment (trackleft) [inline README.md:43]:
```suggestion
to use the latest stable version of the `2.x` branch use `latest-2.x`**
```

File: README.md [hunk 5]
```diff
@@ -1,13 +1,75 @@
...
+
+### Release is tagged on the `main` branch
+All tagged releases on the `main` branch will:
+- Update `latest-5x` bucket
+- Create a new bucket for the version that was released. Example: `5.0.0`
+
+### Release is tagged on the `2.x` branch
+All tagged releases on the `2.x` branch will:
+- Update `latest-2x` bucket
```
Comment (trackleft) [inline README.md:52]:
```suggestion
- Update `latest-2.x` bucket
```

File: README.md [hunk 6]
```diff
@@ -1,13 +1,75 @@
...
+
+**NOTE: `main` is deprecated and will no longer be updated as of `v2.0.20` - to
+use the latest development version of the `2.x` branch use `2.x`**
+
+**NOTE: `latest` is deprecated and will no longer be updated as of `v2.0.20` -
+to use the latest stable version of the `2.x` branch use `latest-2x`**
+
+### Release is tagged on the `main` branch
+All tagged releases on the `main` branch will:
+- Update `latest-5x` bucket
```
Comment (trackleft) [inline README.md:47]:
```suggestion
- Update `latest-5.x` bucket
```

**Review by joeparsons — DISMISSED** [review]

**Review by mmunro-ltrr — DISMISSED** [review]

**Review by mmunro-ltrr — APPROVED** [review]

**Review by joeparsons — APPROVED** [review]

**Pull request merged**